test/bin/wait_images.py

```diff
@@ -0,0 +1,110 @@
+# $ sudo composer-cli compose status --json
+# [
+#     {
+#         "method": "GET",
+#         "path": "/compose/queue",
+#         "status": 200,
+#         "body": {
+#             "new": [
+#                 {
+#                     "blueprint": "rhel-9.2",
+#                     "compose_type": "image-installer",
+#                     "id": "7bdb78bf-c4b6-4f19-80df-073d38fade56",
+#                     "image_size": 0,
+#                     "job_created": 1687466891.2393456,
+#                     "queue_status": "WAITING",
+#                     "version": "0.0.1"
+#                 }
+#             ],
+#             "run": [
+#                 {
+#                     "blueprint": "rhel-9.2",
+#                     "compose_type": "edge-commit",
+#                     "id": "4aa19f32-54e3-42ce-a4ba-cf038a3df91c",
+#                     "image_size": 0,
+#                     "job_created": 1687466889.5383687,
+#                     "job_started": 1687466889.5480232,
+#                     "queue_status": "RUNNING",
+#                     "version": "0.0.1"
+#                 }
+#             ]
+#         }
+#     },
+#     {
+#         "method": "GET",
+#         "path": "/compose/finished",
+#         "status": 200,
+#         "body": {
+#             "finished": []
+#         }
+#     },
+#     {
+#         "method": "GET",
+#         "path": "/compose/failed",
+#         "status": 200,
+#         "body": {
+#             "failed": []
+#         }
+#     }
+# ]
+def flattened_status():
+    result = subprocess.run(STATUS_COMMAND, stdout=subprocess.PIPE)
+    if result.returncode != 0:
+        raise SystemError(f'Status command returned {result.returncode}')
+    status = json.loads(result.stdout)
+    for result_set in status:
+        for state in result_set["body"]:
+            for job in result_set["body"][state]:
+                yield job
+
+
+def main(build_ids):
+    ignore_ids = set()
+    known_ids = set(build_ids)
+    while build_ids:
+        for job in flattened_status():
+            job_id = job["id"]
+            status_text = f'{job_id} {job["compose_type"]} for {job["blueprint"]} - {job["queue_status"]}'
+            if job_id in build_ids:
+                logging.info(status_text)
+                if job["queue_status"] in {"FAILED", "FINISHED"}:
+                    # After a job fails or finishes, stop reporting its status.
+                    build_ids.remove(job_id)
+            elif job_id not in ignore_ids and job_id not in known_ids:
+                # Report any unknown jobs one time, then ignore them.
+                logging.info(f'{status_text} (unknown job)')
+                ignore_ids.add(job_id)
+        if build_ids:
+            time.sleep(30)
+
+
+if __name__ == '__main__':
+    cli_parser = argparse.ArgumentParser(add_help=True)
+    cli_parser.add_argument(
+        'build_id',
+        nargs='+',
+        help="a build id (UUID) from composer",
```

Comment:
This should probably be fixed to denote multiple build_id(s)?

Reply:
The help output shows that the option can be repeated, but the script isn't meant for anyone to use directly. It is called by build_images.sh.